### PR TITLE
PARQUET-2179: [parquet-cpp] Add a test for skipping repeated fields

### PR DESCRIPTION
**Pull request by fatemehp**
Add a test for skipping repeated fields to make it clear that we are skipping values and not records. Also, added some comments to the existing test for Skip of non-repeated values.

### CODE REVIEW

File: cpp/src/parquet/column_reader_test.cc [hunk 1]
```diff
@@ -317,13 +318,64 @@ TEST_F(TestPrimitiveReader, TestInt32FlatRequiredSkip) {
...
+// values and not records.
+TEST_F(TestPrimitiveReader, TestSkipRepeatedField) {
+  // Example schema: message M { repeated int32 b = 1 }
+  max_def_level_ = 1;
+  max_rep_level_ = 1;
+  NodePtr type = schema::Int32("b", Repetition::REPEATED);
+  const ColumnDescriptor descr(type, max_def_level_, max_rep_level_);
+  // Example rows: {}, {[10, 10]}, {[20, 20, 20]}
+  std::vector<int32_t> values = {10, 10, 20, 20, 20};
+  std::vector<int16_t> def_levels{0, 1, 1, 1, 1, 1};
```
Comment (emkornfield) [inline cpp/src/parquet/column_reader_test.cc]: small nit: consistent syntax for initialzation please.

File: cpp/src/parquet/column_reader_test.cc [hunk 2]
```diff
@@ -317,13 +318,64 @@ TEST_F(TestPrimitiveReader, TestInt32FlatRequiredSkip) {
...
+                              Encoding::PLAIN, /*indices=*/{},
+                              /*indices_size=*/0, def_levels, max_def_level_,
+                              rep_levels, max_rep_level_);
+
+  pages_.push_back(std::move(page));
+
+  InitReader(&descr);
+  Int32Reader* reader = static_cast<Int32Reader*>(reader_.get());
+
+  std::vector<int32_t> vresult(4, -1);
```
Comment (emkornfield) [inline cpp/src/parquet/column_reader_test.cc]: nit: could we make make variable names slightly more descriptive?

Comment (fatemehp) [reply cpp/src/parquet/column_reader_test.cc]: Addressed the comments, PTAL.

**Comment by fatemehp** [comment]
@emkornfield could you take a look?

**Review by emkornfield — APPROVED** [review]
will merge when I'm at a computer that I have access to computer with permissions.

**Comment by emkornfield** [comment]
It actually looks like this is PR against your fork?  Do you want to open one against apache/arrow?

**Pull request merged**